mllib/src/main/scala/org/apache/spark/ml/r/LogisticRegressionWrapper.scala

```diff
@@ -84,14 +93,12 @@ private[r] object LogisticRegressionWrapper
 
     val rFormula = new RFormula()
       .setFormula(formula)
-    RWrapperUtils.checkDataColumns(rFormula, data)
+      .setForceIndexLabel(true)
+    checkDataColumns(rFormula, data)
     val rFormulaModel = rFormula.fit(data)
 
-    // get feature names from output schema
-    val schema = rFormulaModel.transform(data).schema
-    val featureAttrs = AttributeGroup.fromStructField(schema(rFormulaModel.getFeaturesCol))
-      .attributes.get
-    val features = featureAttrs.map(_.name.get)
+    // get labels and feature names from output schema
+    val (features, labels) = getFeaturesAndLabels(rFormulaModel, data)
 
     // assemble and fit the pipeline
     val logisticRegression = new LogisticRegression()
```

Comment:
off-topic, but I think it's a bug. We should not allow users pass ```fitIntercept``` to control whether to fit intercept, this should be handled by formula. For example, if users specify formula ```y ~ a + b + c - 1```, then the model should be fitted w/o intercept. Could you please fix this bug as well? Thanks.